src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -237,12 +265,20 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
         }
         else if (!info.image_location.isEmpty())
         {
-            // TODO: Need to make this async like in DefaultVMImageVault
+            QString image_path;
             QTemporaryDir lxd_import_dir{template_path};
 
-            auto image_path = lxd_import_dir.filePath(mp::vault::filename_for(info.image_location));
+            if (query.query_type != Query::Type::LocalFile)
+            {
+                // TODO: Need to make this async like in DefaultVMImageVault
+                image_path = lxd_import_dir.filePath(mp::vault::filename_for(info.image_location));
 
-            url_download_image(info, image_path, monitor);
+                url_download_image(info, image_path, monitor);
+            }
+            else
+            {
+                image_path = mp::vault::copy(source_image.image_path, lxd_import_dir.path());
```

Comment:
Couldn't we avoid this? And just use the original file?

Reply:
I suppose we could, but I have two reasons for doing it this way.
1. One is if the file is a compressed `xz` file, the `post_process_downloaded_image()` function takes care of uncompressing it and will delete the original file.
2. Following on `1.`, it allows commonizing most of the logic between both HTTP & file launches.

Reply:
I suppose it's something we should make common for all the backends - decompress / image conversion / copy should be NOOP if unnecessary.